### ISSUE TITLE
Convert onnx to ncnn with python?

### ISSUE THREAD

**Issue by joeyballentine** [Tencent/ncnn#3548]
## detail | 详细描述 | 詳細な説明

Hi,

Is there a way to convert an ONNX model to NCNN with python? I see no good examples anywhere on the internet and the onnx2ncnn from this repo is c++

My ultimate goal is to be able to pogrammaically convert pytorch models to ncnn, so using convertmodel.com is not something I can use as I want to do it through code 

Do you have any suggestions? 

Thank you

**Comment by caishanli** [comment]
use this [https://convertmodel.com/](url)

**Comment by joeyballentine** [comment]
As I stated previously, my goal is to be able to convert in python code. My end goal is to be able to have a user be able to select a pytorch model, and if they do not have an Nvidia GPU my code can just convert the model to ncnn automatically and use that instead. I cannot do this if I have to make my users go to convertmodel.com.

**Comment by JohnTian** [comment]
@joeyballentine Four months have passed, do u find the solution ?

**Comment by joeyballentine** [comment]
> @joeyballentine Four months have passed, do u find the solution ? 

Yes, the solution is we are rewriting the entirety of onnx2ncnn in python. It's still in progress.

**Comment by JohnTian** [comment]
> > @joeyballentine Four months have passed, do u find the solution ?
> 
> Yes, the solution is we are rewriting the entirety of onnx2ncnn in python. It's still in progress.

 :+1: Good luck!  Is it possible to share the onnx2ncnn python script by create a PR in this repo?

**Comment by joeyballentine** [comment]
> > > @joeyballentine Four months have passed, do u find the solution ?
> > 
> > Yes, the solution is we are rewriting the entirety of onnx2ncnn in python. It's still in progress.
> 
>  :+1: Good luck!  Is it possible to share the onnx2ncnn python script by create a PR in this repo?

That's up to @theflyingzamboni as he's the one doing all the work on it, but I think he's just going to put it in its own separate repo

**Comment by JohnTian** [comment]
> > > > @joeyballentine Four months have passed, do u find the solution ?
> > > 
> > > 
> > > Yes, the solution is we are rewriting the entirety of onnx2ncnn in python. It's still in progress.
> > 
> > 
> > +1 Good luck!  Is it possible to share the onnx2ncnn python script by create a PR in this repo?
> 
> That's up to @theflyingzamboni as he's the one doing all the work on it, but I think he's just going to put it in its own separate repo

OK, Thanks!

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
